output.go

```diff
@@ -184,6 +186,29 @@ func (vmOutput *VMOutput) GetFirstReturnData(asType vm.ReturnDataKind) (interfac
 	return nil, fmt.Errorf("can't interpret return data")
 }
 
+// GetMaxOutputTransferIndex returns the maximum output transfer index
+func (vmOutput *VMOutput) GetNextAvailableOutputTransferIndex() uint32 {
+	maxTransferIndex := -1
+	for _, account := range vmOutput.OutputAccounts {
+		for _, transfer := range account.OutputTransfers {
+			if int(transfer.Index) > maxTransferIndex {
+				maxTransferIndex = int(transfer.Index)
+			}
+		}
+	}
+	return uint32(maxTransferIndex + 1)
+}
+
+// ReindexTransfers from VMOutput
+func (vmOutput *VMOutput) ReindexTransfers(nextIndexProvider NextOutputTransferIndexProvider) {
+	for _, account := range vmOutput.OutputAccounts {
```

Comment:
this is not deterministic. as the iteration on accounts is done in a map. Why is this function needed ?